exercises/chapter7/src/Data/AddressBook/Validation.purs

```diff
@@ -64,16 +66,34 @@ lengthIs _     _   value = pure value
 -- ANCHOR_END: lengthIs
 
 -- ANCHOR: phoneNumberRegex
-phoneNumberRegex :: Either String Regex
-phoneNumberRegex = regex "^\\d{3}-\\d{3}-\\d{4}$" noFlags
+-- | We use `Data.String.Regex.Unsafe.unsafeRegex` here instead of `Data.String.Regex.regex` in order
+-- | to simplify the code.
+-- |
+-- | The safe function has this signature:
+-- |
+-- | ```purescript
+-- | regex :: String -> RegexFlags -> Either String Regex
+-- | ```
+-- |
+-- | which can fail if we pass an invalid regex `String`. Since we construct the string ourselves
+-- | the only reason it would fail is if we were careless. Even in this case it's safe for us to use:
+-- |
+-- | ```purescript
+-- | unsafeRegex :: String -> RegexFlags -> Regex
+-- | ```
+-- |
+-- | instead because the error would be cought at compile time. That's because `phoneNumberRegex` is
+-- | defined as a top level binidng and so it's resolve at compile time. This wouldn't be the case
+-- | if we for example would pass in a `String` that we get from use input.
```

Comment:
I _wish_ we had compile-time checks for expressions with literal arguments. Might be able to add this feature in the future.

```suggestion
-- | which can fail if passed an invalid regex `String`. This failing case is worth
-- | checking for at runtime when working with a user-provided regex `String`.
-- | But in our case, we provide a hardcoded regex string, so we can use this
-- | slightly more convenient "unsafe" version that throws an exception:
-- |
-- | ```purescript
-- | unsafeRegex :: String -> RegexFlags -> Regex
-- | ```
-- |
-- | We can achieve a bit more safety by binding our `Regex` values at the top level,
-- | so any potential runtime exceptions are thrown as soon as our application starts.
-- | This is better than defining these values in a local context, where the error may
-- | not be encountered until later on during application execution.
```